test/nodejs-bundle/tcp+websockets+webrtc-star.js

```diff
@@ -91,57 +91,32 @@ describe('TCP + WebSockets + WebRTCStar', () => {
       (cb) => nodeTCP.stop(cb),
       (cb) => nodeWS.stop(cb),
       (cb) => nodeWStar.stop(cb),
-      (cb) => ss.stop(done)
+      (cb) => ss.stop(cb)
     ], done)
   })
 
+  function check (otherNode, done, muxed, peers) {
+    let i = 1;
+    [nodeAll, otherNode].forEach((node) => {
+      expect(Object.keys(node.peerBook.getAll())).to.have.length(i-- ? peers : 1)
+      expect(Object.keys(node.swarm.muxedConns)).to.have.length(muxed)
+    })
+    done()
+  }
+
   it('nodeAll.dial nodeTCP using PeerInfo', (done) => {
     nodeAll.dial(nodeTCP.peerInfo, (err) => {
       expect(err).to.not.exist()
 
       // Some time for Identify to finish
-      setTimeout(check, 500)
-
-      function check () {
-        parallel([
-          (cb) => {
-            const peers = nodeAll.peerBook.getAll()
-            expect(Object.keys(peers)).to.have.length(1)
-            expect(Object.keys(nodeAll.swarm.muxedConns)).to.have.length(1)
-            cb()
-          },
-          (cb) => {
-            const peers = nodeTCP.peerBook.getAll()
-            expect(Object.keys(peers)).to.have.length(1)
-            expect(Object.keys(nodeTCP.swarm.muxedConns)).to.have.length(1)
-            cb()
-          }
-        ], done)
-      }
+      setTimeout(check, 500, nodeTCP, done, 1, 1)
```

Comment:
Please use instead:

`setTimeout(() => check(nodeTCP, 1, 1, done), 500`

And update the function arguments respectively (rule: callback always comes last)